### PR TITLE
Update linter

### PR DESCRIPTION
**Pull request by danlaine**
Updates the linters and linter settings to match avalanchego, and updates the code to pass the linter.

There are some places where it wasn't clear to me whether to handle a previously unhandled error, or to ignore it. For example, if deletion of a file fails, should the error be propagated or should the program ignore it? I will leave it to you all to decide.

Also, if you feel that this repo shouldn't have certain linters/settings used in avalanchego, then by all means feel free to remove them.

### CODE REVIEW

File: cmd/subnetcmd/delete.go
Comment (holisticode) [inline cmd/subnetcmd/delete.go:53]: We will want to review all of these `Stat`s for the above issue.

File: cmd/subnetcmd/publish.go
Comment (holisticode) [inline cmd/subnetcmd/publish.go:323]: Please add a quick comment how we got here (`err != nil` which happened pretty early)

File: cmd/subnetcmd/publish.go
Comment (holisticode) [inline cmd/subnetcmd/publish.go:325]: This is not from your PR, but could you pls add a `TODO` here? It's a bit dangerous we are assigning to a global down here.

File: pkg/config/config.go
Comment (holisticode) [inline pkg/config/config.go:18]: Probably

File: pkg/prompts/prompts.go
Comment (holisticode) [inline pkg/prompts/prompts.go:283]: Huh, why does the linter not complain here? We always use `ID` for such labels?

File: pkg/ux/output.go
Comment (holisticode) [inline pkg/ux/output.go:35]: Huh, this looks a bit ugly, do we really need to do this?

File: cmd/subnetcmd/delete.go
Comment (danlaine) [reply cmd/subnetcmd/delete.go:38]: Like I said in the PR description
```
There are some places where it wasn't clear to me whether to handle a previously unhandled error, or to ignore it.
```
Totally up to you guys to decide what the right thing to do is in cases like this one

File: pkg/prompts/prompts.go
Comment (danlaine) [reply pkg/prompts/prompts.go:283]: It's ok with "id" and "ID", but it doesn't like "Id" and it doesn't like local scoped things starting with capital letters:

```
pkg/prompts/prompts.go:283:2: unexported-naming: the symbol IDStr is local, its name should start with a lowercase letter (revive)
```

File: pkg/config/config.go
Comment (danlaine) [reply pkg/config/config.go:18]: I'll leave that to you guys to do -- I don't think the `Config` struct is actually used anywhere (it's empty) and can be removed altogether. Maybe I'm missing something, though

File: pkg/ux/output.go
Comment (danlaine) [reply pkg/ux/output.go]: changed linting rules so this isn't needed

File: cmd/subnetcmd/publish.go
Comment (danlaine) [reply cmd/subnetcmd/publish.go]: added comment on this function

File: cmd/subnetcmd/publish.go
Comment (danlaine) [reply cmd/subnetcmd/publish.go]: refactored this function -- see if it's alright with you now

File: cmd/subnetcmd/delete.go
Comment (danlaine) [reply cmd/subnetcmd/delete.go:53]: Same as above -- totally up to you guys to decide what the right thing to do is in cases like this one.

Would you mind handling such cases? You'd know better than me whether we want to ignore or propagate errors in various places.

File: cmd/subnetcmd/delete.go
Comment (holisticode) [reply cmd/subnetcmd/delete.go:38]: OK, but the point is that this is now changing the semantics inside your PR. 

So my suggestion here would be to ignore the error if the file does not exist and not return an non-nil `err`, requiring a change in this PR. @dasconnor @felipemadero

File: pkg/config/config.go
Comment (holisticode) [reply pkg/config/config.go:18]: Yeah was just a comment from my end agreeing with you. No action on your end required.

File: cmd/subnetcmd/delete.go
Comment (danlaine) [reply cmd/subnetcmd/delete.go:38]: Sure, do you want me to remove all changes in this PR that deal with the success of file deletions?

Comment (felipemadero) [reply cmd/subnetcmd/delete.go:38]: I think we can approve this if keeping the same sematics. That is not returning the new errors, instead just make a log.

Comment (dasconnor) [reply cmd/subnetcmd/delete.go:38]: Yeah, I think we want to keep the file deletion logic, that was intentional, mostly we don't care if we can't delete something because it's not actually there.

**Review by dasconnor — COMMENTED** [review]
I'm good to merge this as soon as we revert the file deletion changes

**Review by dasconnor — APPROVED** [review]

**Review by felipemadero — APPROVED** [review]
LGTM

**Pull request merged**
